packages/language-picker/src/language-picker.tsx

```diff
@@ -0,0 +1,186 @@
+			const sharedTerritories = intersection( lg.subTerritories, selectedLanguage.territories );
+
+			if ( sharedTerritories.length > 0 ) {
+				return lg;
+			}
+
+			return false;
+		} )?.id ?? defaultLananguageGroupId
+	);
+};
+
+const LanguagePicker = ( {
+	onSelectLanguage,
+	languages,
+	languageGroups,
+	selectedLanguage,
+	localizedLanguageNames,
+}: Props ) => {
+	const [ filter, setFilter ] = useState(
+		findBestDefaultLanguageGroupId( selectedLanguage, languageGroups, languageGroups[ 0 ].id )
+	);
+
+	const [ search, setSearch ] = useState( '' );
+
+	const getFilteredLanguages = () => {
+		switch ( filter ) {
+			case 'popular':
+				return languages
+					.filter( ( language ) => language.popular )
+					.sort( ( a, b ) => ( a.popular as number ) - ( b.popular as number ) );
+			default: {
+				const languageGroup = languageGroups.find( ( l ) => l.id === filter );
+				const subTerritories = languageGroup ? languageGroup.subTerritories : [];
+				return languages
+					.filter( ( language ) =>
+						language.territories.some( ( t ) => subTerritories?.includes( t ) )
+					)
+					.sort( ( a, b ) => a.name.localeCompare( b.name ) );
+			}
+		}
+	};
+
+	const renderCategoryButtons = () => {
+		return languageGroups.map( ( languageGroup ) => {
+			const isSelected = filter === languageGroup.id;
+
+			const onClick = () => {
+				setFilter( languageGroup.id );
+			};
+
+			return (
+				<Button
+					key={ languageGroup.id }
+					onClick={ onClick }
+					className="language-picker-component__language-group"
+				>
+					<span className={ isSelected ? 'is-selected' : '' }>{ languageGroup.name() }</span>
+				</Button>
+			);
+		} );
+	};
+
+	const shouldDisplayRegions = ! search;
+
+	const languagesToRender = search
+		? getSearchedLanguages( languages, search, localizedLanguageNames )
+		: getFilteredLanguages();
+
+	const selectControlOptions = languageGroups.map( ( lg ) => ( { key: lg.id, name: lg.name() } ) );
+
+	return (
+		<div className="language-picker-component">
+			<div className="language-picker-component__heading">
+				<div className="language-picker-component__title wp-brand-font">
+					{ __( 'Select a language' ) }
+				</div>
+				<div className="language-picker-component__search">
+					<CustomSelectControl
+						label={ __( 'regions' ) }
+						hideLabelFromVision
+						value={ selectControlOptions.find( ( option ) => option.key === filter ) }
+						options={ selectControlOptions }
+						onChange={ ( { selectedItem } ) => selectedItem && setFilter( selectedItem.key ) }
+					/>
+					<Search
+						onSearch={ setSearch }
+						pinned
+						fitsContainer
+						placeholder={ __( 'Search languages…' ) }
+					/>
+				</div>
+			</div>
+			<div className="language-picker-component__labels">
+				{ shouldDisplayRegions ? (
+					<>
+						<div className="language-picker-component__regions-label">{ __( 'regions' ) }</div>
+						<div className="language-picker-component__languages-label">{ __( 'language' ) }</div>
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Language', { context: 'logged-out-homepage'} )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/use-lp-in-calypso)